### ISSUE TITLE
#undef REQUIRE_PLUGIN is the only thing you can do.

### ISSUE THREAD

**Issue by eyal282** [sbpp/sourcebans-pp#715]
In short, if you don't undef, then it says sourcecomms.smx isn't loaded for some bizzare reason.

**Comment by sneak-it** [comment]
https://github.com/sbpp/sourcebans-pp/blob/5331ab9d759308adef97d6bd026ba3d650aa6411/game/addons/sourcemod/scripting/include/sourcecomms.inc#L110-L128

Seems like this is already properly set.

If you are writing a plugin utilizing sourcecomms natives and you wish for them to be optional, you need to specify that the plugin is optional like so:

```
#undef REQUIRE_PLUGIN
#tryinclude <sourcecomms>
```

Whatever you are doing inside of your plugin needs to make sure it is attempting to use sourcecomms natives after all plugins are loaded - `OnAllPluginsLoaded` is available for your use.

TL;DR - This issue doesn't seem to be an issue with sourcecomms, moreso with how you are utilizing natives in the plugin you are writing.

**Comment by Groruk** [comment]
Just like Sneak mentioned, this is likely caused on your end.

**Comment by eyal282** [comment]
I think you misunderstood. the absence of #undef REQUIRE_PLUGIN will break the plugin, regardless if it's loaded or not.

**Comment by asherkin** [comment]
Your error mentions `sourcecomms.smx`, but the plugin binary here is `sbpp_comms.smx`, I think your issue is coming from something else (possibly an outdated include file on your development machine?)

The registration here is all right on the sbpp_comms side, the names are correct and the plugin library being registered matches the include - so all the SM expectations are met for dependences to work correctly.
